libs/evcs-service/src/test/java/uk/gov/di/ipv/core/library/pact/ContractTest.java

```diff
@@ -0,0 +1,430 @@
+                                                                metadata.numberType(
+                                                                        "timestampMs",
+                                                                        1711721297123L);
+                                                                metadata.stringType(
+                                                                        "txmaEventId",
+                                                                        "1a116fe7-2ff9-4f7c-940d-d55fa7d03d66");
+                                                            });
+                                                });
+                                    });
+                        })
+                .build();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "validRetrieveVcRequestReturnsUsersVcWith200")
+    void testRetrieveVcRequestReturnsUsersVcWith200(MockServer mockServer) throws Exception {
+        // Under Test
+        EvcsClient evcsClient = new EvcsClient(mockConfigService);
+        var evcsGetUserVCsDto =
+                evcsClient.getUserVcs(TEST_USER_ID, TEST_EVCS_ACCESS_TOKEN, VC_STATES_FOR_QUERY);
+
+        // Assertions
+        assertEquals(1, evcsGetUserVCsDto.vcs().size());
+    }
+
+    @Pact(provider = "EvcsProvider", consumer = "EvcsConsumer")
+    public RequestResponsePact invalidRetrieveVcRequestReturnsWith400(PactDslWithProvider builder) {
+        return builder.given("EVCS client exist")
+                .given("VCS exist")
+                .given("EVC has users VC")
+                .uponReceiving("A request to get VCS")
+                .path("/vcs/" + INVALID_USER_ID)
+                .query("state=" + PENDING_RETURN)
+                .method("GET")
+                .headers(
+                        Map.of(
+                                "x-api-key",
+                                EVCS_API_KEY,
+                                "Authorization",
+                                "Bearer " + TEST_EVCS_ACCESS_TOKEN))
+                .willRespondWith()
+                .status(400)
+                .toPact();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "invalidRetrieveVcRequestReturnsWith400")
+    void testRetrieveVcRequestReturns400(MockServer mockServer) {
+        // Under Test
+        EvcsClient evcsClient = new EvcsClient(mockConfigService);
+        assertThrows(
+                EvcsServiceException.class,
+                () -> {
+                    evcsClient.getUserVcs(
+                            INVALID_USER_ID, TEST_EVCS_ACCESS_TOKEN, VC_STATES_FOR_QUERY);
+                });
+    }
+
+    @Pact(provider = "EvcsProvider", consumer = "EvcsConsumer")
+    public RequestResponsePact invalidRetrieveVcRequestReturnsWith404(PactDslWithProvider builder) {
+        return builder.given("EVCS client exist")
+                .given("VCS exist")
+                .given("EVC has users VC")
```

Comment:
What's the difference between this case and the one above?

Reply:
Its just diff status code we receive when its not finding user, also 404 not error on our end. I think I need to update the given example.